mirrord/layer/tests/outgoing.rs

```diff
@@ -88,12 +88,20 @@ async fn outgoing_udp(dylib_path: &PathBuf) {
 /// 2. Connects to the remote peer
 /// 3. Sends some data
 /// 4. Expects the peer to send the same data back
-#[rstest]
-#[tokio::test]
-#[timeout(Duration::from_secs(10))]
-async fn outgoing_tcp(dylib_path: &PathBuf) {
+async fn outgoing_tcp_logic(with_config: Option<&str>, dylib_path: &PathBuf, config_dir: &PathBuf) {
+    let config = with_config
+        .and_then(|config| {
+            let mut config_path = config_dir.clone();
+            config_path.push(config);
+            Some(config_path)
+        })
+        .unwrap_or_default();
+
+    let config = config.to_str().unwrap();
+    let config = (!config.is_empty()).then_some(config);
```

Comment:
Isn't this the same?
```suggestion
    let config = with_config
        .map(|config| {
            let mut config_path = config_dir.clone();
            config_path.push(config);
            config_path
        })
        .map(|path_buf| path_buf.to_str().unwrap());
```

Reply:
updated this suggestion.